tsl/src/bgw_policy/job_api.c

```diff
@@ -25,6 +25,30 @@
 
 #define ALTER_JOB_NUM_COLS 8
 
+/*
+ * Check configuration for a job type.
+ */
+static void
+job_config_check(Name proc_schema, Name proc_name, Jsonb *config)
```

Comment:
If you would use the PolicyConfig approach outlined above, this would looks something like:

```


typedef enum PolicyType
{
  POLICY_TYPE_REORDER,
  POLICY_TYPE_RETENTION,
  POLICY_TYPE_CUSTOM,
} PolicyType;

typedef struct PolicyConfig
{
    PolicyType type;
    const char *name;	
} PolicyConfig;

typedef struct PolicyDef
{
    const char *schema;
    const char *name;
    PolicyConfig *(*parse_func)(const Jsonb *config);
} PolicyDef;

static PolicyConfig *
policy_reorder_parse(const Jsonb *config)
{
	...
}


static PolicyConfig *
policy_retention_parse(const Jsonb *config)
{
   ....
}

/* This is the canonical place to define new policies */
static const PolicyDef policies[] = {
    [POLICY_TYPE_REORDER] =
	{
		.schema = INTERNAL_SCHEMA_NAME,
		.name = "policy_reorder",
		.parse_func = policy_reorder_parse,
	},
	[POLICY_TYPE_RETENTION] =
	{
		.schema = INTERNAL_SCHEMA_NAME,
		.name = "policy_retention",
		.parse_func = policy_retention_parse,
	}
};

PolicyConfig *
policy_parse(const char *schema, const char *name, const Jsonb *json)
{
	int i;

	for (i = 0; i < lengthof(policies); i++)
	{
		const PolicyDef *pdef = &policies[i];
		
		if (strcmp(pdef->schema, schema) == 0 &&
			strcmp(pdef->name, name) == 0)
			return pdef->parse_func(json);
	}

	elog(ERROR, "no policy matching ....");
}
	
```